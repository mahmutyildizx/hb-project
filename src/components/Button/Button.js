import React from "react";
import cn from "classnames";
import styles from "./Button.module.scss";

function Button({ name, variation, onClick, disable }) {
  const { button, disabled_class } = styles;
  return (
    <button
      onClick={onClick}
      disabled={disable}
      className={cn(button, styles[variation], disable && disabled_class)}
    >
      <span>{name}</span>
    </button>
  );
}

export default Button;
