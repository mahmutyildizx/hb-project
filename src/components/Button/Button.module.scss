.buttonContainer {
  width: 100%;
  min-width: 120px;
  padding: 14px 20px;
  background-color: #ffffff;
  color: #b0b0b0;
  border: 1px solid #b0b0b0;
  border-radius: 4px;
  font-size: 16px;
  font-weight: 500;
  line-height: 24px;
  position: relative;
  cursor: pointer;

  &.buttonIcon {
    display: flex;
    align-items: center;
    justify-content: space-between;
  }

  .quantityBadge {
    position: absolute;
    z-index: 2;
    width: 20px;
    height: 20px;
    border-radius: 50%;
    padding: 10px;
    top: -10px;
    right: -10px;
    display: flex;
    align-items: center;
    justify-content: center;
    background-color: #ff6000;
    color: #ffffff;
    font-size: 12px;
  }

  &.orange {
    background-color: #ff6000;
    border: 1px solid transparent;
    border-radius: 4px;

    span {
      color: #ff6000;
      font-size: 12px;
    }

    &:disabled {
      cursor: not-allowed;
      background-color: gray;

      span {
        color: gray;
      }
    }
  }

  &.accept {
    background-color: #90d659;

    span {
      color: #fff;
    }
  }

  &.cancel {
    background-color: #d65959;

    span {
      color: #fff;
    }
  }

  &.cartRemove {
    background-color: #fff;
    border: 1px solid #f90000;
    border-radius: 4px;
    min-width: 48px;
    width: 48px;
    height: 20px;
    line-height: 20px;
    display: flex;
    align-items: center;
    justify-content: center;

    span {
      color: #f90000;
      font-size: 12px;
    }

    &:hover {
      background-color: #f90000;

      span {
        color: #fff;
      }
    }
  }
}
