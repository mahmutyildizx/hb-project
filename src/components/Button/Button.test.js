import React from "react";
import { render, cleanup, screen } from "@testing-library/react";
import preloadAll from "jest-next-dynamic";
import Button from "./Button";

afterEach(cleanup);

beforeAll(async () => {
  await preloadAll();
});

test("should render", () => {
  const { container } = render(<Button />);
  expect(container.firstChild).toHaveClass("buttonContainer");
});

test("should match snapshot", () => {
  const { container } = render(<Button />);
  expect(container).toMatchSnapshot();
});

test("Button label rendered as text", () => {
  render(<Button name="Test Button" />);
  expect(screen.getByText("Test Button")).toBeInTheDocument();
});