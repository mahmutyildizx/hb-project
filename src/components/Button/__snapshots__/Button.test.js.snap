// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  <button
    class="buttonContainer"
  >
    <span />
  </button>
</div>
`;

exports[`should render 1`] = `
<div>
  <button
    class="buttonContainer"
  >
    <span />
  </button>
</div>
`;
